### ISSUE TITLE
installed repo packages confused with identically named AUR packages

### ISSUE THREAD

**Issue by steven-omaha** [Morganamilo/paru#383]
### Affected Version
`paru v1.6.0.r43.gba35318 +git - libalpm v13.0.0`
Build from the latest HEAD, ba35318d61ade7e37da73e6f9daf9f2a1de1e3ad.

### Description
I have `zfs-linux` installed from the `archzfs-testing` repository.  I just wanted to upgrade. paru thinks that zfs-linux and zfs-linux-lts originate from the AUR, instead of from archzfs-testing. If I proceed with the update, it lets me review the corresponding PKGBUILDs from the AUR. It should recognize that I do not have the AUR version installed, but a repository version.

This is a new bug. Used to work until very recently.

### Output
**Include the FULL output of any relevant commands/configs**

```sh



$ paru
no new news
:: Paketdatenbanken werden synchronisiert …
 core ist aktuell
 extra ist aktuell
 community ist aktuell
 archzfs-testing ist aktuell
 panthera ist aktuell
:: Looking for AUR upgrades
:: Looking for devel upgrades
9 core/linux              5.12.5.arch1-1         -> 5.12.6.arch1-1
8 core/linux-headers      5.12.5.arch1-1         -> 5.12.6.arch1-1
7 core/linux-lts          5.10.38-1              -> 5.10.39-1
6 core/linux-lts-headers  5.10.38-1              -> 5.10.39-1
5 extra/nvidia            465.31-3               -> 465.31-4
4 extra/nvidia-lts        1:465.31-2             -> 1:465.31-3
3 aur/zfs-linux           2.0.4_5.12.5.arch1.1-1 -> 2.0.4_5.12.6.arch1.1-1
2 aur/zfs-linux-lts       2.0.4_5.10.38.1-1      -> 2.0.4_5.10.39.1-1
1 aur/zoom                5.6.4-1                -> 5.6.6-1
:: Packages to exclude (eg: 1 2 3, 1-3):
::

# confirming this then clones zfs-linux and zfs-linux-lts from the AUR.
----------------------------------------------------------------------


$ sudo pacman -Syu
:: Paketdatenbanken werden synchronisiert …
 core ist aktuell
 extra ist aktuell
 community ist aktuell
 archzfs-testing ist aktuell
 panthera ist aktuell
:: Vollständige Systemaktualisierung wird gestartet …
Abhängigkeiten werden aufgelöst …
Nach in Konflikt stehenden Paketen wird gesucht …

Paket (8)                      Alte Version            Neue Version            Netto-Veränderung  Größe des Downloads

core/linux                     5.12.5.arch1-1          5.12.6.arch1-1                   0,02 MiB            94,88 MiB
core/linux-headers             5.12.5.arch1-1          5.12.6.arch1-1                   0,00 MiB            23,32 MiB
core/linux-lts                 5.10.38-1               5.10.39-1                        0,03 MiB            74,26 MiB
core/linux-lts-headers         5.10.38-1               5.10.39-1                        0,00 MiB            22,65 MiB
extra/nvidia                   465.31-3                465.31-4                        -0,01 MiB            23,41 MiB
extra/nvidia-lts               1:465.31-2              1:465.31-3                      -0,01 MiB            23,28 MiB
archzfs-testing/zfs-linux      2.0.4_5.12.5.arch1.1-1  2.0.4_5.12.6.arch1.1-1           0,03 MiB            13,32 MiB
archzfs-testing/zfs-linux-lts  2.0.4_5.10.38.1-1       2.0.4_5.10.39.1-1               -0,04 MiB            13,05 MiB

Gesamtgröße des Downloads:             288,18 MiB
Gesamtgröße der installierten Pakete:  508,43 MiB
Größendifferenz der Aktualisierung:      0,03 MiB

:: Installation fortsetzen? [J/n]




----------------------------------------------------------------------
#
# /etc/pacman.conf
#
# See the pacman.conf(5) manpage for option and repository directives

#
# GENERAL OPTIONS
#
[options]
# The following paths are commented out with their default values listed.
# If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
CacheDir     = /zroot/cache/packages/pacman/pkg
#LogFile     = /var/log/pacman.log
#GPGDir      = /etc/pacman.d/gnupg/
#HookDir     = /etc/pacman.d/hooks/
HoldPkg     = pacman-git glibc
#XferCommand = /usr/bin/curl -L -C - -f -o %o %u
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#CleanMethod = KeepInstalled
Architecture = auto

# Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
#IgnorePkg   =
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

# Misc options
#UseSyslog
Color
#NoProgressBar
CheckSpace
VerbosePkgLists
ParallelDownloads = 5

# By default, pacman accepts packages signed by keys that its local keyring
# trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel    = Required DatabaseOptional
LocalFileSigLevel = Optional
#RemoteFileSigLevel = Required

# NOTE: You must run `pacman-key --init` before first using pacman; the local
# keyring can then be populated with the keys of all official Arch Linux
# packagers with `pacman-key --populate archlinux`.

#
# REPOSITORIES
#   - can be defined here or included from another file
#   - pacman will search repositories in the order defined here
#   - local/custom mirrors can be added here or in separate files
#   - repositories listed first will take precedence when packages
#     have identical names, regardless of version number
#   - URLs will have $repo replaced by the name of the current repo
#   - URLs will have $arch replaced by the name of the architecture
#
# Repository entries are of the format:
#       [repo-name]
#       Server = ServerName
#       Include = IncludePath
#
# The header [repo-name] is crucial - it must be present and
# uncommented to enable the repo.
#

# The testing repositories are disabled by default. To enable, uncomment the
# repo name header and Include lines. You can add preferred servers immediately
# after the header, and they will be used before the default mirrors.

#[testing]
#Include = /etc/pacman.d/mirrorlist

[core]
Include = /etc/pacman.d/mirrorlist

[extra]
Include = /etc/pacman.d/mirrorlist

#[community-testing]
#Include = /etc/pacman.d/mirrorlist

[community]
Include = /etc/pacman.d/mirrorlist

[archzfs-testing]
SigLevel = Required
Server = http://archzfs.com/$repo/x86_64

[panthera]
SigLevel = Optional
Server = myprivateserver

# If you want to run 32 bit applications on your x86_64 system,
# enable the multilib repositories as required here.

#[multilib-testing]
#Include = /etc/pacman.d/mirrorlist

#[multilib]
#Include = /etc/pacman.d/mirrorlist

# An example of a custom package repository.  See the pacman manpage for
# tips on creating your own repositories.
#[custom]
#SigLevel = Optional TrustAll
#Server = file:///home/custompkgs

----------------------------------------------------------------------
paru.conf
------
[options]
BatchInstall
BottomUp
CleanAfter
CombinedUpgrade
NewsOnUpgrade
RemoveMake
SudoLoop
UpgradeMenu
Devel


```

**Comment by Morganamilo** [comment]
The way testing works on archlinux is packages are inserted in testing, then when they are ready they are *moved* to core. This means the package no longer exists in testing and this would break if you only had testing enabled.

I have no idea if the zfs repo works this way but it's my only hunch. I can't reproduce and don't see what else it could be.

**Comment by steven-omaha** [comment]
IMO this is not related to the archzfs-testing repo. If I change pacman.conf to use archzfs instead of archzfs-testing, I get the same error from paru.

paru wants to do this:
```
 3 aur/zfs-linux                           2.0.4_5.12.5.arch1.1-1 -> 2.0.4_5.12.6.arch1.1-1
 2 aur/zfs-linux-lts                       2.0.4_5.10.38.1-1      -> 2.0.4_5.10.39.1-1
```

pacman wants to do this:
```
archzfs/zfs-linux                       2.0.4_5.12.5.arch1.1-1  2.0.4_5.12.6.arch1.1-1           0,03 MiB            13,32 MiB
archzfs/zfs-linux-lts                   2.0.4_5.10.38.1-1       2.0.4_5.10.39.1-1               -0,04 MiB            13,05 MiB
```

To me it seems paru incorrectly assumes that I have AUR/zfs-linux installed, instead of the repo version.

```
$ pacman -Si zfs-linux | head -n3
Repositorium             : archzfs
Name                     : zfs-linux
Version                  : 2.0.4_5.12.6.arch1.1-1

$ paru -Si zfs-linux | grep -i reposit
Repository      : aur
```

It appears paru does not take archzfs into consideration when looking for packages.

In comparison, yay finds this: 
```
$ yay -Si zfs-linux
Repositorium             : archzfs-testing
Name                     : zfs-linux
Version                  : 2.0.4_5.12.6.arch1.1-1
```

And yay would like to do this:
```
 3  archzfs-testing/zfs-linux               2.0.4_5.12.5.arch1.1-1 -> 2.0.4_5.12.6.arch1.1-1
 2  archzfs-testing/zfs-linux-lts           2.0.4_5.10.38.1-1      -> 2.0.4_5.10.39.1-1
```

My yay config:
```json
{
	"aururl": "https://aur.archlinux.org",
	"buildDir": "$HOME/.cache/yay",
	"absdir": "$HOME/.cache/yay/abs",
	"editor": "nvim",
	"editorflags": "",
	"makepkgbin": "/usr/bin/makepkg",
	"makepkgconf": "",
	"pacmanbin": "/usr/bin/pacman",
	"pacmanconf": "/etc/pacman.conf",
	"redownload": "no",
	"rebuild": "tree",
	"batchinstall": true,
	"answerclean": "none",
	"answerdiff": "all",
	"answeredit": "",
	"answerupgrade": "",
	"gitbin": "git",
	"gpgbin": "gpg",
	"gpgflags": "",
	"mflags": "",
	"sortby": "votes",
	"searchby": "name-desc",
	"gitflags": "",
	"removemake": "yes",
	"sudobin": "sudo",
	"sudoflags": "",
	"requestsplitn": 150,
	"sortmode": 0,
	"completionrefreshtime": 1,
	"sudoloop": true,
	"timeupdate": false,
	"devel": false,
	"cleanAfter": true,
	"provides": true,
	"pgpfetch": true,
	"upgrademenu": true,
	"cleanmenu": false,
	"diffmenu": true,
	"editmenu": true,
	"combinedupgrade": true,
	"useask": false
}

```

**Comment by Morganamilo** [comment]
#386 has made me think it's an issue with signing being enabled now. I'll revert that for now and it should fix this.

**Comment by steven-omaha** [comment]
Thanks. I will report back when the next kernel update happens.

**Comment by steven-omaha** [comment]
TLDR: 23dfaa2 fixed the issue.

I just tested this scenario on a different machine, when I was going to apply the updates mentioned in the original report.

I had paru-1.6.1-1 installed, build on May 22. paru intended to apply the updates from archzfs-testing.

I could not test paru-bin since it does not work with pacman-git.

paru-git compiled from current HEAD also applies archzfs-testing updates correctly.

Then I changed the PKGBUILD to checkout ba35318 before building. Indeed, now paru intends to fetch the updates from the AUR instead of archzfs-testing.
